### ISSUE TITLE
Allow for multiple extendr modules in one Rust library

### ISSUE THREAD

**Issue by clauswilke** [extendr/extendr#151]
I'd like to be able to have multiple extendr modules in one Rust library. Something like this:

```rust
#[extendr]
fn add(x: f64, y: f64) -> f64 {
    x + y
}

#[extendr]
fn sub(x: f64, y: f64) -> f64 {
    x - y
}

extendr_module! {
    mod add;
    fn add;
}

extendr_module! {
    mod sub;
    fn sub;
}
```
This is particularly useful for extendrtests, but may have applications more broadly.

This works in principle, except for one issue: If we try to call the routine registration functions for the two modules, the most recent registration always overwrites any previous registration for the same library. So, for example, this does not work:
```C
void R_init_extendrtests(void *dll) {
  R_init_add_extendr(dll);
  R_init_sub_extendr(dll);
}
```
Only the `sub` module gets registered.

However, we could implement something like the following which would work:
```C
const char* mods[] = { "add", "sub" };

void R_init_extendrtests(void *dll) {
  R_init_extendr(dll, mods, 2);
}
```
I.e., we provide a generic init function that registers all modules it is given.

This feature could be implemented in addition to the current initialization scheme, so people who have only a single module can call that module's function directly but people who have multiple modules can register all of them at once.

**Comment by clauswilke** [comment]
The api I outlined here may not work, as it may not be that easy (or even possible) to call the metadata functions for the different modules by module name provided as a string. However, an alternative approach will definitely be possible: Call Rust functions from C to retrieve pointers to the Rust data structures with the metadata, and then call a Rust function to register all functions in that metadata.

**Comment by clauswilke** [comment]
This seems to be addressed with #160.
